### PR TITLE
Minimum Octave to 5.1, remove workarounds

### PR DESCRIPTION
**Pull request by cbm755**
I'm a bit unsure whether to just leave some of this...  @alexvong1995 what do you think?

At least the stuff in `octsympy_tests.m` can certainly go.

Fixes Issue #1074.

### CODE REVIEW

File: inst/private/python_ipc_driver.m [hunk 1]
```diff
@@ -33,10 +33,10 @@
 
   %% version check
   if exist('OCTAVE_VERSION', 'builtin')
-    if (compare_versions (OCTAVE_VERSION (), '4.2', '<'))
+    if (compare_versions (OCTAVE_VERSION (), '5.1', '<'))
       fprintf(['\n********************************************************************\n' ...
                'Your Octave version is %s but Octsympy is currently untested on\n' ...
-               'anything before 4.2.\n\n'], OCTAVE_VERSION ())
+               'anything before 5.1.\n\n'], OCTAVE_VERSION ())
```
Comment (cbm755) [inline inst/private/python_ipc_driver.m:39]: this feels right to me, but less certain about actively removing workarounds.

Comment (alexvong1995) [reply inst/private/python_ipc_driver.m:39]: I think we can also safely remove workarounds in ` %!test`.

File: inst/@sym/private/do_highbyte_escapes.m [hunk 2]
```diff
@@ -48,14 +48,7 @@
...
   % get the two-char hex numbers make them into bytes
-  if (exist ('OCTAVE_VERSION', 'builtin') && ...
-      compare_versions (OCTAVE_VERSION (), '4.3.0', '<'))
-    % Bug on old Octave: https://savannah.gnu.org/bugs/?49659
-    dec = char(hex2dec(NM.hex));
-  else
-    % roughly 3-4 times slower than the above
-    dec = char (hex2dec (struct2cell (NM)));
-  end
+  dec = char (hex2dec (struct2cell (NM)));
```
Comment (alexvong1995) [inline inst/@sym/private/do_highbyte_escapes.m:51]: According to https://savannah.gnu.org/bugs/?49659, the bug was fixed in version 4.4 (2018-04-30). One can check that the fix is in version 5.1 https://hg.savannah.gnu.org/hgweb/octave/file/d05d6eebde10/libinterp/corefcn/regexp.cc as well. LGTM!

File: inst/private/do_highbyte_escapes.m [hunk 3]
```diff
@@ -48,14 +48,7 @@
...
   % get the two-char hex numbers make them into bytes
-  if (exist ('OCTAVE_VERSION', 'builtin') && ...
-      compare_versions (OCTAVE_VERSION (), '4.3.0', '<'))
-    % Bug on old Octave: https://savannah.gnu.org/bugs/?49659
-    dec = char(hex2dec(NM.hex));
-  else
-    % roughly 3-4 times slower than the above
-    dec = char (hex2dec (struct2cell (NM)));
-  end
+  dec = char (hex2dec (struct2cell (NM)));
```
Comment (alexvong1995) [inline inst/private/do_highbyte_escapes.m:51]: The change is the same as above. LGTM!

File: inst/@sym/display.m [hunk 4]
```diff
@@ -75,14 +75,9 @@ function display(x)
...
+  if (exist('OCTAVE_VERSION', 'builtin'))
     [fmt, spacing] = format();
     loose = strcmp (spacing, 'loose');
-  elseif (exist('OCTAVE_VERSION', 'builtin') && ...
-      compare_versions (OCTAVE_VERSION (), '4.0.0', '>='))
-    % Octave 4.1 dropped (temporarily?) the get(0,...) approach
-    loose = eval('! __compactformat__ ()');
   else
     % Matlab and Octave < 4
     loose = strcmp(get(0, 'FormatSpacing'), 'loose');
```
Comment (alexvong1995) [inline inst/@sym/display.m:83]: As per the docstring of `[FORMAT, FORMATSPACING] = format` in Octave 5.1 https://hg.savannah.gnu.org/hgweb/octave/file/d05d6eebde10/libinterp/corefcn/pr-output.cc, `loose` is the default format for displaying matrices. We should be able to drop the `eval` stuff w/o any issues.

**Review by alexvong1995 — APPROVED** [review]
I didn't look deeply into changes in `octsympy_tests.m` and `%!test`. Well, those tests will fail anyway if things don't work. I didn't see any problem in removing other workarounds. It's all good!

**Pull request merged**